include/lsst/afw/typehandling/SimpleGenericMap.h

```diff
@@ -105,7 +104,10 @@ class SimpleGenericMap final : public MutableGenericMap<K> {
 protected:
     ConstValueReference unsafeLookup(K key) const override {
         try {
-            return _storage.at(key);
+            return std::visit(
+                [](auto const & v) { return ConstValueReference(std::cref(v)); },
```

Comment:
Again, it seems strange that this is necessary. Both `reference_wrapper` and `variant` have implicit constructors, so why can't you implicitly convert a `StorableType` to a `ConstValueReference`?